front/app/containers/ProjectsShowPage/timeline/setPhaseURL.ts

```diff
@@ -0,0 +1,26 @@
+import { IPhaseData } from 'services/phases';
+import { IProjectData } from 'services/projects';
+import { Locale } from 'typings';
+import { isNilOrError } from 'utils/helperUtils';
+
+export default function setPhaseUrl(
+  phase: IPhaseData | null,
+  phases: Error | IPhaseData[] | null | undefined,
+  project: IProjectData | null | undefined,
+  locale: Error | Locale | null | undefined
+) {
+  if (
+    phase === null ||
+    isNilOrError(phases) ||
+    !project ||
+    isNilOrError(locale)
+  ) {
+    return;
+  }
+
+  const phaseNumber = phases.findIndex((p) => p.id === phase.id) + 1;
```

Comment:
The above would allow you to use more descriptive names here too:

`const phaseNumber = phases.findIndex((phase) => selectedPhase.id === phase.id) + 1;`

By convention, we also use the var coming from the array looping (`phase `here) at the right side of the operator, but that's reaaally nitpicking. You don't need to change that, just a FYI. :-)